torch/cuda/__init__.py

```diff
@@ -308,7 +308,6 @@ def stream(stream):
 def device_count():
     """Returns the number of GPUs available."""
     if is_available():
-        _lazy_init()
```

Comment:
Won't `is_available()` initialize CUDA anyway?

Reply:
As of the current implementation no. And `torch._C._cuda_getDeviceCount()` does not require it to be initialized (again from tests from python).